R/ide.R

```diff
@@ -47,11 +47,12 @@ validateConnectUrl <- function(url, certificate = NULL) {
   retry <- TRUE
   while (retry) {
     tryCatch({
-      httpResponse <- GET(parseHttpUrl(url),
+      response <- GET(parseHttpUrl(url),
                           list(certificate = certificate),
                           settingsEndpoint,
                           timeout = getOption("rsconnect.http.timeout", 10))
 
+      httpResponse <- attr(response, "httpResponse")
```

Comment:
Note to self: It confused me at first when the result from `GET` was not a response object. If callers to `GET` need to react to its status, maybe we should think about augmenting the lower-level `result` object with a `json` field.

```r
response <- GET(...)
result <- response$json
status <- response$status
```

That's a pretty broad change, and excessive for the majority of cases, which do not want to react to status codes.